File: python-03/ex0/ft_command_quest.py
#!/usr/bin/env python3

import sys

print("=== Command Quest ===")

argc = len(sys.argv)

if argc < 2:
    print("No arguments provided!")
    print(f"Program name: {sys.argv[0]}")
else:
    print(f"Program name: {sys.argv[0]}")
    print(f"Arguments received: {argc - 1}")

    arg_counter = 1
    for arg_value in sys.argv[1:]:
        print(f"Argument {arg_counter}: {arg_value}")
        arg_counter += 1

print(f"Total arguments: {argc}")

File: cpp-03/ex0/ft_command_quest.cpp
#include <iostream>

int main(int argc, char** argv)
{
    int i = 0;
    std::cout << "=== Command Quest ===" << std::endl;
    if (argc < 2)
    {
        std::cout << "No arguments provided!" << std::endl;
        std::cout << "Program name: " << *argv << std::endl;
    }
    else
    {
     std::cout << "Program name: " << *argv << std::endl;
     std::cout << "Arguments received: " << argc - 1 << std::endl;
     while(*++argv && i++ < argc)
         std::cout << "Argument " << i << ": " << *argv << std::endl;
    }
    std::cout << "Total arguments: " << argc << std::endl;
     return 0;
}
